tritium-registry/build.gradle

```diff
@@ -0,0 +1,13 @@
+apply from: "${rootDir}/gradle/publish-jars.gradle"
+
+dependencies {
+    processor "org.immutables:value"
+
+    compile 'com.palantir.safe-logging:safe-logging'
+    compile 'io.dropwizard.metrics:metrics-core'
```

Comment:
This is the thing I poked at in remoting.

Reply:
as in you want interfaces for all of these things?

Reply:
@uschi2000's concern was that this would end up making impossible diamond deps. I'm not hard over on the split vs. not, except that if we're really going to try to get this to be the 'api' kit then it shouldn't depend on anything but the apis?

Reply:
I've split out an interface for the TaggedMetricRegistry so if the remaining concern is depending on metrics-core I'd argue it's fine to depend on this as an api dependency. It's currently used by 60k projects https://github.com/search?q=com.codahale.metrics.MetricRegistry&type=Code&utf8=%E2%9C%93 and things like spark will break if they break their api.

Reply:
not sure I understand what this split buys us, then

Reply:
interface split or project split? the project split means remoting doesn't depend on tritium-api or tritium-core. I'm indifferent to the interface split, but don't think it hurts.

Reply:
I'm not sure how much indirection we want to buy. I think this PR gets us enough so that folks that want the minimum can depend on `tritium-registry`, and in apps depend on the full thing and force deps to avoid any diamond issues.